src/lib/list/selection-list.spec.ts

```diff
@@ -0,0 +1,234 @@
+import {async, TestBed} from '@angular/core/testing';
+import {Component} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {MdSelectionList, MdListOption, MdListModule} from './index';
+import {createKeyboardEvent} from '../core/testing/event-objects';
+import {UP_ARROW, DOWN_ARROW, SPACE} from '../core/keyboard/keycodes';
+
+
+describe('SelectionList and ListOption', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MdListModule],
+      declarations: [
+        SelectionListWithListOptions,
+        SelectionListWithCheckbocPositionAfter,
+        SelectionListWithListDisabled
+      ],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  it('should add and remove focus class on focus/blur', () => {
+    let fixture = TestBed.createComponent(SelectionListWithListOptions);
+    fixture.detectChanges();
+    let listOption = fixture.debugElement.query(By.directive(MdListOption));
+    let listItemEl = fixture.debugElement.query(By.css('.mat-list-item'));
+
+    expect(listItemEl.nativeElement.classList).not.toContain('mat-list-item-focus');
+
+    listOption.componentInstance._handleFocus();
+    fixture.detectChanges();
+    expect(listItemEl.nativeElement.classList).toContain('mat-list-item-focus');
+
+    listOption.componentInstance._handleBlur();
+    fixture.detectChanges();
+    expect(listItemEl.nativeElement.classList).not.toContain('mat-list-item-focus');
+  });
+
+  it('should be able to dispatch selected one item', () => {
+    let fixture = TestBed.createComponent(SelectionListWithListOptions);
+    fixture.detectChanges();
+    let listOption = fixture.debugElement.queryAll(By.directive(MdListOption));
+    let selectionList = fixture.debugElement.query(By.directive(MdSelectionList));
+    let testListItem = listOption[2].injector.get<MdListOption>(MdListOption);
+
+    testListItem.toggle();
+    fixture.detectChanges();
+
+    let selectList = selectionList.injector.get<MdSelectionList>(MdSelectionList).selectedOptions;
+
+    expect(selectList.selected.length).toBe(1);
```

Comment:
:+1:  Added check `expect(selectList.selected.length).toBe(0); `before the `toggle`